src/java/org/apache/cassandra/service/accord/AccordKeyspace.java

```diff
@@ -0,0 +1,843 @@
+                                                                         long timestamp,
+                                                                         int nowInSec,
+                                                                         StoredNavigableMap<K, V> map,
+                                                                         Function<K, ByteBuffer> serializeKey,
+                                                                         Function<V, ByteBuffer> serializeVal)
+    {
+        if (map.wasCleared())
+        {
+            if (!map.hasAdditions())
+            {
+                builder.addComplexDeletion(column, new DeletionTime(timestamp, nowInSec));
+                return;
+            }
+            else
+                builder.addComplexDeletion(column, new DeletionTime(timestamp - 1, nowInSec));
+        }
+
+        map.forEachAddition((k, v) -> builder.addCell(live(column, timestamp, serializeVal.apply(v), CellPath.create(serializeKey.apply(k)))));
+
+        if (!map.wasCleared())
+            map.forEachDeletion(k -> builder.addCell(tombstone(column, timestamp, nowInSec, CellPath.create(serializeKey.apply(k)))));
+    }
+
+    private static <T extends Comparable<?>> void addStoredSetChanges(Row.Builder builder,
+                                                                      ColumnMetadata column,
+                                                                      long timestamp,
+                                                                      int nowInSec,
+                                                                      StoredSet<T, ?> map,
+                                                                      Function<T, ByteBuffer> serialize)
+    {
+        if (map.wasCleared())
+        {
+            if (!map.hasAdditions())
+            {
+                builder.addComplexDeletion(column, new DeletionTime(timestamp, nowInSec));
+                return;
+            }
+            else
+                builder.addComplexDeletion(column, new DeletionTime(timestamp - 1, nowInSec));
+        }
+
+        map.forEachAddition(i -> builder.addCell(live(column, timestamp, EMPTY_BYTE_BUFFER, CellPath.create(serialize.apply(i)))));
+
+        if (!map.wasCleared())
+            map.forEachDeletion(k -> builder.addCell(tombstone(column, timestamp, nowInSec, CellPath.create(serialize.apply(k)))));
+    }
+
+    public static Mutation getCommandMutation(AccordCommand command, long timestampMicros)
+    {
+        try
+        {
+            Preconditions.checkArgument(command.hasModifications());
+
+            // TODO: convert to byte arrays
+            ValueAccessor<ByteBuffer> accessor = ByteBufferAccessor.instance;
+
+            Row.Builder builder = BTreeRow.unsortedBuilder();
+            builder.newRow(Clustering.EMPTY);
+            int nowInSeconds = (int) TimeUnit.MICROSECONDS.toSeconds(timestampMicros);
+            int version = MessagingService.current_version;
```

Comment:
similar to previous comment... it would be best if we didn't rely on the messaging version as this is not for messaging but for value versioning in the table... so if we move version into the `blob` then we can version each field in isolation.